### PR TITLE
Migrate escape room challenge

### PR DESCRIPTION
**Pull request by kriskowal**
This moves and dusts off the escape room challenge from https://github.com/endojs/endo/tree/master/packages/ses/demos

### CODE REVIEW

**Review by erights — COMMENTED** [review]
FWIW, the parts I am familiar with LGTM. I defer to @warner for the rest.

File: public/challenge/index.css
Comment (warner) [inline public/challenge/index.css:1]: (best $10 I ever spent, although for future reference I'll note that the site is now allowing commercial use for free)

File: public/challenge/index.js
Comment (warner) [inline public/challenge/index.js]: typos

File: public/challenge/index.js
Comment (warner) [inline public/challenge/index.js]: This only builds the Compartment once per page-load, right? How does that work with `enableDateNow`/`disableDateNow` merely modifying `endowments` and changing the `window.history`, but not reloading the page? I haven't tried to interact with the demo to confirm, but it seems like the attacker's Compartment would continue to get the original authority.

This might require that `submitProgram` create a new Compartment each time.

**Review by warner — APPROVED** [review]
Looks good in general, but if that Compartment issue is real, please fix that before landing.

**Comment by warner** [comment]
Oh, also maybe make the Attacker Code box smaller, or move the Attack/Stop buttons above the box, because with the current layout, I need a pretty big window to see the Secret/Guess boxes change as I push the Attack button. I like that the window scrolls to show the Guess box, but 1: it's surprising to have the button I just clicked jump away from me, and 2: it doesn't show the Secret window at the same time.

(It'd probably be ok to make the Secret/Guess boxes shorter, I think the aspect ratio matches the original WarGames graphics, but not a lot of us remember that movie and would appreciate the accuracy..)

File: public/challenge/index.js
Comment (warner) [reply public/challenge/index.js]: Indeed, I tested this by:
* load the challenge page
* click "Timing Channel Frustrated", which 1: loads the attack code, 2: leaves the "Expose" checkbox unchecked, 3: leaves the URL alone (no `?dateNow=enabled`)
* now click-on the Expose checkbox, which 1: checks the checkbox, 2: changes the URL to add `?dateNow=enabled`
* now click "Attack"
* observe: the attack fails (guesses all zeros and then repeats)

Comment (kriskowal) [reply public/challenge/index.js]: Good catch.

File: public/challenge/index.js
Comment (kriskowal) [reply public/challenge/index.js]: Weird.

File: src/content/docs/challenge.md
Comment (mhofman) [inline src/content/docs/challenge.md:9]:
```suggestion
[Macguffin](https://en.wikipedia.org/wiki/Macguffin), in this case a random
```

Comment (kriskowal) [reply src/content/docs/challenge.md:9]: Pushed a patch. Thanks.

**Pull request merged**
